app/src/main/java/org/mifos/mobile/ui/beneficiary_detail/BeneficiaryDetailScreen.kt

```diff
@@ -0,0 +1,259 @@
+    navigateBack: () -> Unit,
+    updateBeneficiaryClicked: () -> Unit,
+    deleteBeneficiary: (id: Long?) -> Unit,
+) {
+    var openDropdown by rememberSaveable { mutableStateOf(false) }
+    var showAlert by rememberSaveable { mutableStateOf(false) }
+    val context = LocalContext.current
+    var deleteButtonClicked by remember { mutableStateOf(false) }
+    var showBeneficiaryNullError by rememberSaveable {
+        mutableStateOf(true)
+    }
+
+    if (showAlert) {
+        MifosAlertDialog(onDismissRequest = {
+            showAlert = !showAlert
+        },
+            dismissText = context.getString(R.string.cancel),
+            onConfirmation = {
+                showAlert = !showAlert
+                deleteButtonClicked = true
+                deleteBeneficiary.invoke(beneficiary?.id?.toLong())
+            },
+            confirmationText = context.getString(R.string.delete),
+            dialogTitle = context.getString(R.string.delete_beneficiary),
+            dialogText = context.getString(R.string.delete_beneficiary_confirmation) + "?"
+        )
+    }
+
+    Scaffold(topBar = {
+        BeneficiaryDetailTopAppBar(navigateBack = navigateBack,
+            updateBeneficiaryClicked = updateBeneficiaryClicked,
+            updateDropdownValue = { value ->
+                openDropdown = value
+                openDropdown
+            },
+            showAlert = {
+                showAlert = true
+            })
+    }) {
+        Box(
+            modifier = Modifier.padding(it)
+        ) {
+            BeneficiaryDetailContent(
+                beneficiary = beneficiary
+            )
+
+            when (uiState) {
+                BeneficiaryUiState.Initial -> Unit
+
+                BeneficiaryUiState.Loading -> {
+                    MifosProgressIndicator(
+                        modifier = Modifier
+                            .fillMaxSize()
+                            .background(MaterialTheme.colorScheme.background.copy(0.8f))
+                    )
+                }
+
+                BeneficiaryUiState.DeletedSuccessfully -> {
+                    Toast.makeText(
+                        context,
+                        stringResource(id = R.string.beneficiary_deleted_successfully),
+                        Toast.LENGTH_SHORT
+                    ).show()
+                    navigateBack.invoke()
+                }
+
+                is BeneficiaryUiState.ShowError -> {
+                    if (deleteButtonClicked) {
+                        Toast.makeText(context, uiState.message, Toast.LENGTH_SHORT).show()
+                        deleteButtonClicked = false
+                    }
+                    if (showBeneficiaryNullError) {
+                        Toast.makeText(context, uiState.message, Toast.LENGTH_SHORT).show()
+                        showBeneficiaryNullError = false
+                    }
+                }
+
+                else -> Log.d("Error uiState", "Undesired $uiState\")")
```

Comment:
remove else, we only have 4 states , so no need of else.

Reply:
`BeneficiaryUiState` has more than 4 states. Removing `else` will give us an error on `when(uiState)`